src/components/directed-requester.js

```diff
@@ -0,0 +1,42 @@
+const Requester = require('./requester');
+
+module.exports = class DirectedRequester extends Requester {
+    constructor(advertisement, discoveryOptions) {
+        super(advertisement, discoveryOptions);
+        this.sock.send = this.socketSend.bind(this);
+    }
+
+    socketSend(...args) {
+        // Enqueue if no socks connected yet
+        const requesterSock = this.sock;
+        if (!requesterSock.socks || !requesterSock.socks.length) return this.sock.enqueue(args);
+
+        const data = args[0];
+        if (!data.responderId) {
+            throw new Error('send() needs a "responderId" property when using DirectedRequester');
```

Comment:
```suggestion
            throw new Error('send() needs a "responderId" property in the request body when using DirectedRequester');
```

Reply:
Also, do you think it would make sense to rename this to `__responderId` much like we do with `__timeout` and `__room` properties?